analyzers/src/SonarAnalyzer.CSharp/Rules/UnconditionalJumpStatement.cs

```diff
@@ -75,15 +74,34 @@ private class LoopWalker : LoopWalkerBase<StatementSyntax, SyntaxKind>
                 SyntaxKind.CatchClause
             };
 
-            public LoopWalker(SyntaxNodeAnalysisContext context, ISet<SyntaxKind> loopStatements)
-                : base(context, loopStatements)
+            public LoopWalker(SyntaxNodeAnalysisContext context, ISet<SyntaxKind> loopStatements) : base(context, loopStatements) { }
+            public override void Visit()
             {
+                var csWalker = new CsLoopwalker(this);
+                csWalker.SafeVisit(rootExpression);
             }
 
-            public override void Visit()
+            protected override bool IsPropertyAccess(StatementSyntax node) =>
+                node.DescendantNodes().Any(x => x.IsKind(SyntaxKind.IdentifierName)
+                                                && x is IdentifierNameSyntax identifier
+                                                && semanticModel.GetSymbolInfo(identifier).Symbol is { } symbol
+                                                && symbol.Kind == SymbolKind.Property);
+
+            protected override bool IsAnyKind(SyntaxNode node, ISet<SyntaxKind> syntaxKinds) => node.IsAnyKind(syntaxKinds);
+            protected override bool IsReturnStatement(SyntaxNode node) => node.IsKind(SyntaxKind.ReturnStatement);
```

Comment:
Formatting (EOL between them, as they are not properties and they are not trivial)
```suggestion
            protected override bool IsAnyKind(SyntaxNode node, ISet<SyntaxKind> syntaxKinds) => 
                node.IsAnyKind(syntaxKinds);
            
            protected override bool IsReturnStatement(SyntaxNode node) => 
                node.IsKind(SyntaxKind.ReturnStatement);
```